### ISSUE TITLE
Inline union assigned to type alias produces error

### ISSUE THREAD

**Issue by tcobbs-bentley** [python/mypy#14302]
**Bug Report**
The following code produces an error when using mypy as a linter in Visual Studio Code:

```python
Test = tuple[str, str] | str
```

(Using pylint as a linter, the above code does not produce an errror.)

The following (equivalent, AFAIK) code does not produce an error when using mypy as a linter:

```python
Test = Union[tuple[str, str], str]
```

For reasons that aren't really clear, the following code does not produce an error (even though it seems similar to the code that produces an error):

```python
SQLiteDescription = tuple[tuple[str, None, None, None, None, None, None], ...] | Any
```

For reference, the error (in Visual Studio Code) is this:

```json
[{
	"resource": "/Users/travis/Dev/itwin/mobile-samples/cross-platform/startuptimes/startuptimes.py",
	"owner": "python",
	"code": "error",
	"severity": 8,
	"message": "Type application has too many types (1 expected)  [misc]",
	"source": "mypy",
	"startLineNumber": 23,
	"startColumn": 8,
	"endLineNumber": 23,
	"endColumn": 8
}]
```

(A clear and concise description of what the bug is.)

**To Reproduce**

1. Install Visual Studio Code
2. Install Microsoft's Python plugin
3. Open a Python file.
4. Use the Python: Select Linter command (from F1 command palette) to select mypy as the linter.
5. Install mypy if prompted to do so.
6. Include the following code in your Pyton file:

```python
Test = tuple[str, str] | str
```

**Expected Behavior**

The code should be accepted by mypy.

**Actual Behavior**

mypy gives the following error:

    Type application has too many types (1 expected)  [misc]

**Your Environment**

- Mypy version used: 0.991
- Mypy command-line flags: Unknown (linter in VS Code)
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: 3.10.6

**Comment by AlexWaygood** [comment]
(It's already fixed on mypy `master`; the fix will be included in the next release, mypy 1.0.)

**Comment by tcobbs-bentley** [comment]
Sorry, I didn't search closed issues.

**Comment by AlexWaygood** [comment]
No worries! Follow #13685 for updates on when the next release will be coming.
